File: java/src/test/java/problems1_100/MergeTwoSortedListsTest.java
package problems1_100;

import concepts.SinglyListNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class MergeTwoSortedListsTest extends SampledTest {
    @Test
    public void test() {
        for (MergeTwoSortedLists solution : MergeTwoSortedLists.values()) {
            for (MergeTwoSortedListsSample sample : getSamples(MergeTwoSortedListsSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.mergeTwoLists(
                            SinglyListNode.of(sample.input.list1),
                            SinglyListNode.of(sample.input.list2)
                        )
                    ).isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/MergeTwoSortedListsTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import concepts.SinglyListNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class MergeTwoSortedListsTest : SampledTest() {
    @Test
    fun test() =
        MergeTwoSortedLists.entries.forEach { solution ->
            getSamples(Array<MergeTwoSortedListsSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(
                            solution.mergeTwoLists(
                                SinglyListNode.of(*input.list1),
                                SinglyListNode.of(*input.list2),
                            ),
                        ).isEqualTo(SinglyListNode.of(*output))
                }
        }
}
